Update macOS-Development-Setup.rst

PhDittmann opened the pull request:
- add Xcode App Store link
- BASH -> ZSH (default)
- make homebrew paths generic

audrow commented on source/Installation/Alternatives/macOS-Development-Setup.rst:
```diff
@@ -29,7 +29,7 @@ You need the following things installed to build ROS 2:
    **Xcode**
 
-   * If you don't already have it installed, install Xcode.
+   * If you don't already have it installed, install [Xcode](https://apps.apple.com/de/app/xcode/id497799835).
```
```suggestion
   * If you don't already have it installed, install [Xcode](https://apps.apple.com/app/xcode/id497799835).
```

This also works but doesn't link to xcode in German, I think.

PhDittmann replied:
You're right. I removed the locale and pushed it again.

These changes should also work for the other versions.

audrow reviewed (commented):
I haven't tested this locally, but these changes look good to me.

I suggested one small change.

I also think this PR should target rolling and then be backported to Iron and Humble (if relevant).

audrow commented:
@mergifyio rebase

audrow commented:
@PhDittmann I retargeted this PR to rolling and tried rebasing it since there were a bunch of unnecessary commits. Can you fix this, maybe by cherry-picking your commit onto rolling and then pushing to this branch? Otherwise, I can do it, but I may be slower to get to it.

PhDittmann commented:
I stashed my commit, pulled the latest official rolling and put it on top of it.

PhDittmann committed:
Update macOS-Development-Setup.rst

    - add Xcode App Store link
    - BASH -> ZSH (default)
    - make homebrew paths generic
